numpy/core/_exceptions.py

```diff
@@ -106,13 +116,23 @@ def __init__(self, ufunc, casting, from_, to, i):
         self.out_i = i
 
     def __str__(self):
-        # only show the number if more than one output exists
-        i_str = "{} ".format(self.out_i) if self.ufunc.nout != 1 else ""
+        if self.ufunc.nout > 1:
+            from_outs = ", ".join(["out{}={}".format(i+1, f.name) for i, f in enumerate(self.from_) if i>=self.ufunc.nin])
+            to_outs = ", ".join(["out{}={}".format(i+1, t.name) for i, t in enumerate(self.to) if i>=self.ufunc.nin])
```

Comment:
This doesn't produce a legal call, it should be:
```suggestion
            to_outs = "out=({})".format(", ".join(t.name for t in self.to[self.ufunc.nin:]))
```

Reply:
(same for the other three lines)

Reply:
This comment still applies